test_runner/src/main/kotlin/ftl/json/SavedMatrix.kt

```diff
@@ -97,9 +103,24 @@ private fun SavedMatrix.updateProperties(newMatrix: TestMatrix) = copy(
     clientDetails = newMatrix.getClientDetails(),
     gcsPathWithoutRootBucket = newMatrix.getGcsPathWithoutRootBucket(),
     gcsRootBucket = newMatrix.getGcsRootBucket(),
-    webLinkWithoutExecutionDetails = newMatrix.webLinkWithoutExecutionDetails()
+    webLinkWithoutExecutionDetails = newMatrix.webLinkWithoutExecutionDetails(),
+    appFileName = newMatrix.extractAppFileName() ?: fallbackAppName
 )
 
+private fun TestMatrix.extractAppFileName() = testSpecification?.run {
+    listOf(
+        androidInstrumentationTest,
+        androidTestLoop,
+        androidRoboTest,
+        iosXcTest,
+        iosTestLoop
+    )
+        .firstOrNull { it != null }
```

Comment:
Strange formatting to have a random endline character?
Personal preference or auto corrected?

Reply:
`formatKotlin` did that, I'm not happy with it neither but who am I to challenge it :D
Being serious, I'll think of another formatting that is inline with kotlin format task